sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -979,16 +979,26 @@ private void validateReplicaAddresses(String collectionRid, AddressInformation[]
                                         .stream()
                                         .map(Uri::getURIAsString)
                                         .collect(Collectors.toList()));
+
+                minConnectionsRequiredForEndpoint = this.connectionPolicy.getMinConnectionPoolSizePerEndpoint();
             }
 
             for (Uri addressToBeValidated : addressesNeedToValidation) {
 
+                // replica validation should be triggered for an address with Unknown health status only when
+                // the address is used by a container / collection which was part of the warm up flow
+                if (addressToBeValidated.getHealthStatus() == Uri.HealthStatus.Unknown) {
+                    if (!this.proactiveOpenConnectionsProcessor.isCollectionRidUnderOpenConnectionsFlow(collectionRid)) {
```

Comment:
We can probably merge these two conditions, right?

if (addressToBeValidated == Unknown && !isCollectionUnderOpenConnectionsFlow()) {
continue;
}

Also, what if the replica is not in an Unknown status? We are still triggering the replica validation, right?

Reply:
`UnhealthyPending` is the other replica health status for which replica validation is triggered.

Reply:
Merged `if` check in next iteration.